Framework/Crystal/src/IndexPeaks.cpp

```diff
@@ -500,9 +506,26 @@ void IndexPeaks::exec() {
     std::unordered_map<int, std::vector<IPeak *>> peaksPerRun;
     for (int i = 0; i < args.workspace->getNumberPeaks(); i++)
       peaksPerRun[args.workspace->getPeak(i).getRunNumber()].emplace_back(args.workspace->getPeakPtr(i));
-    if (peaksPerRun.size() < 2) {
-      g_log.warning("Peaks from only one run exist but CommonUBForAll=True so peaks will be indexed with an optimised "
-                    "UB which will not be saved in the workspace.");
+    if (peaksPerRun.size() == 1) {
+      if (args.updateUB) {
+        // Save the optimized UB to the workspace
+        auto peaks = peaksPerRun.begin()->second;
+        std::vector<V3D> qSample(peaks.size());
+        std::generate(std::begin(qSample), std::end(qSample),
```

Comment:
This code seems to be a bit of duplication from `indexPeaks` - it looks like the optimized UB is used to overwrite the `sampleUB` passed to `indexPeaks`
https://github.com/mantidproject/mantid/blob/d0b5a5646b5d5ae0d8c1f8fb97d6bc156117a13c/Framework/Crystal/src/IndexPeaks.cpp#L269-L278
So I think the UB could be set inside this branch 
https://github.com/mantidproject/mantid/blob/d0b5a5646b5d5ae0d8c1f8fb97d6bc156117a13c/Framework/Crystal/src/IndexPeaks.cpp#L498-L515
if `UpdateUB=True` and `args.workspace->getNumberPeaks() == 1` what do you think?

Reply:
@RichardWaiteSTFC what do you think here?